hw/ip/kmac/data/kmac.hjson

```diff
@@ -0,0 +1,340 @@
+                KMAC/SHA3 begins its operation.
+
+                If the mode is cSHAKE, before receiving the message, the
+                hashing logic processes Function name string N and
+                customization input string S first. If KMAC mode is enabled,
+                additionally it processes secret key block.
+                '''
+        } // f: start
+        { bits: "1"
+          name: "process"
+          desc: '''If writes 1 into this field when KMAC/SHA3 began its
+                operation and received the entire message, it computes the
+                digest or signing.
+                '''
+        } // f: process
+        { bits: "2"
+          name: "run"
+          desc: '''The `run` field is used in the sponge squeezing stage.
+                It triggers the keccak round logic to run full 24 rounds.
+                This is optional and used when software needs more digest bits
+                than the keccak rate.
+
+                It only affects when the kmac/sha3 operation is completed.
+                '''
+        } // f: run
+        { bits: "3"
+          name: "done"
+          desc: '''If writes 1 into this field when KMAC/SHA3 absorbing is
+                completed. KMAC/SHA3 hashing engine clears internal varaibles
+                and goes back to Idle state for next command.
+                '''
+        } // f: done
+      ]
+    } // R: CMD
+    { name: "STATUS"
+      desc: '''KMAC/SHA3 Status register.'''
+      swaccess: "ro"
+      hwaccess: "hwo"
+      hwext: "true"
+      fields: [
+        { bits: "0"
+          name: "sha3_idle"
+          desc: "If 1, SHA3 hashing engine is in idle state."
+        }
+        { bits: "1"
+          name: "sha3_absorb"
+          desc: "If 1, SHA3 is receiving message stream and processing it"
+        }
+        { bits: "2"
+          name: "sha3_squeeze"
+          desc: '''If 1, SHA3 completes sponge absorbing stage.
+                In this stage, SW can manually run the hashing engine.
+                '''
+        }
+        { bits: "12:8"
+          name: "fifo_depth"
+          desc: "Message FIFO entry count"
+        }
+        { bits: "14"
+          name: "fifo_empty"
+          desc: "Message FIFO Empty indicator"
+          resval: "1"
+        }
+        { bits: "15"
+          name: "fifo_full"
+          desc: "Message FIFO Full indicator"
+        }
+      ]
+    } // R: STATUS
+    { multireg: {
+        name: "KEY"
+        desc: '''KMAC Secret Key
+
+              KMAC secret key can be up to 512 bit.
+              Order of the secret key is:
+              key[512:0] = {KEY0, KEY1, KEY2, ... , KEY15};
+
+              The registers are allowed to be updated when the engine is in Idle state.
+              If the engine computes the hash, it discards any attempts to update the secret keys
+              and report an error.
+              '''
+        count: "NumWordsKey"
+        cname: "KMAC"
+        hwext: "true"
```

Comment:
btw i dont think you have to make this one a `hwext`.  You could create a `regen` that controls this register, and have that one be software clear and hardware set.  Could save your some typing later :)

Reply:
Unfortnuately, secret key case, we need to store shared key. If SW writes two share key into KMAC/SHA3, then regen will help .

Reply:
sorry what do you mean? Will the secret key from keymgr also be stored here? Is that what you mean? (that does make sense).

Reply:
oooo i see what you're saying. I actually think software will prefer to maintain the key in masked form all the time. That way they protect the key no matter where it goes.

If they want to use an unmasked key, they can always write the unmasked version in one, and 0 in the other.  So I think it's okay to put that requirement on them.  We can poll their opinions though :)